python_modules/dagster/dagster/_core/definitions/asset_checks.py

```diff
@@ -127,3 +141,46 @@ def get_attributes_dict(self) -> Dict[str, Any]:
             specs=self._specs,
             input_output_props=self._input_output_props,
         )
+
+
+@experimental
+def build_blocking_asset_check(
+    asset: "AssetsDefinition",
+    checks: Sequence[AssetChecksDefinition],
+):
+    from dagster import In, graph_asset, op, Output
+
+    # key_prefix=asset.key[0:-1] if len(asset.key) > 1 else None,
+    asset_key = "blocking_" + asset.key.path[-1]
```

Comment:
why do we need to add the prefix? It should just inherit the old one

Reply:
conflict with the names for the graph and the asset op. Changed to just rename the graph, but maybe should rename the op instead

Reply:
I'm confused. Shouldn't there just be one asset key is existence at all?

Reply:
changed the op name, that's safer

Reply:
> I'm confused. Shouldn't there just be one asset key is existence at all?

Yeah. It's a conflict that the Op created by `@asset` had the same name as the graph we're creating. We can just rename the Op